aws_lambda_powertools/utilities/data_classes/cognito_user_pool_event.py

```diff
@@ -575,13 +575,10 @@ def scopes_to_suppress(self, value: list[str]):
 
 class ClaimsAndScopeOverrideDetails(GroupConfigurationBase):
     @property
-    def id_token_generation(self) -> TokenClaimsAndScopeOverrideDetails | None:
-        id_token_generation_details = self._data.get("idTokenGeneration")
-        return (
-            None
-            if id_token_generation_details is None
-            else TokenClaimsAndScopeOverrideDetails(id_token_generation_details)
-        )
+    def id_token_generation(self) -> TokenClaimsAndScopeOverrideDetails:
+        if self._data.get("idTokenGeneration") is None:
+            self._data["idTokenGeneration"] = {}
+        return TokenClaimsAndScopeOverrideDetails(self._data["idTokenGeneration"])
```

Comment:
```suggestion
    def id_token_generation(self) -> TokenClaimsAndScopeOverrideDetails:
        return TokenClaimsAndScopeOverrideDetails(self._data.get("idTokenGeneration", {}))
```

Could this be improved? What do you think, @nc-dirknilius?

I think we can change this in all the code and make it cleaner to read.

Reply:
@leandrodamascena actually, no. This would introduce a similar bug what I fixed in this PR.

Key difference:

The actual code here modifies `self._data` by setting `idTokenGeneration` to `{}` if it doesn't exist
Your suggested code using `.get("idTokenGeneration", {})` returns a new empty dict without modifying `self._data`. Meaning, setting data via `id_token_generation` would have no effect in the resulting dict and therefore would not work in the Lambda.

Reply:
Hey @nc-dirknilius yep, thanks a lot for this feedback, make completely sense!